furusiya/model/systems/system.py

```diff
@@ -9,7 +9,8 @@ def set(self, owner, component):
         self.component_dict[owner] = component
 
     def remove(self, owner):
-        del self.component_dict[owner]
+        if owner in self.component_dict:
```

Comment:
Shouldn't you check if it's in `self.component_dict.keys()`, and *not* `self.component_dict`?

Reply:
This broke a test that tried to delete a property that didn't exist. It wouldn't make sense to check if it doesn't exist, and then delete it.